### ISSUE TITLE
WorkAsync method supporting parallel calls

### ISSUE THREAD

**Issue by rappen** [MscrmTools/XrmToolBox#505]
Being able to perform long-running tasks using async behavior introduced by `WorkAsync` method and `WorkAsyncInfo` class is great.
But launching several tasks simultaneously causes the yellow info panel to stay on top of the window, even after the last task has completed.
Trying to chain async calls by starting a new task at the end of a `PostWorkCallBack` method gives the same effect with the info panel.

I hope this could be solved some way to really make use of the possibility to execute tasks async / in the background.
Of course, when simultaneous tasks are running, it might not be obvious which tasks message should be shown in the info panel, but perhaps just showing the last started one is enough.

Note: Except for the problem with the info panel, running async tasks in parallel works just fine using the `WorkAsync` method!

Perhaps @shytikov or @daryllabar might be able to do something with this? Pretty please? :)

**Comment by daryllabar** [comment]
This sounds very similar to https://github.com/MscrmTools/XrmToolBox/pull/235.  Do you have a plugin that has this issue currently I could test with?

**Comment by daryllabar** [comment]
I think I see it.  There is a single _worker object for the pluginControlBase class.  It is not thread safe.  But even worse, it only has a single _infoPanel that get's stepped on each time WorkAsync is called.  So you call PluginControlBase.WorkAsync() twice in a row, and two panels get created, but when either the first task or second task is completed, it actually closes the second panel, and leaves the first open.

I think the solution is to create a new worker every single time WorkAsync is called, and update SetWorkingMessage to update the most recently created worker.

**Comment by daryllabar** [comment]
PR Submitted!

**Comment by rappen** [comment]
Thanks!!
And yes, very similar to #235. I thought I had reported this before too, but couldn't find it.
Plugin to test with (not public yet, partly due to this problem) https://github.com/Innofactor/XrmToolBox.DateTimeBehaviorUtility
1. Load Attributes
2. Check a few of them
3. Click Analyze (top right)

**Comment by rappen** [comment]
Tried it from my end - initial tests are very positive! 👍

**Comment by daryllabar** [comment]
I was able to edit a local plugin to to throw 3 WorkAsync Calls, each taking a second, and it worked just fine.

**Comment by MscrmTools** [comment]
I guess this has been fixed in November release

**Comment by rappen** [comment]
October release! :)

**Comment by MscrmTools** [comment]
Yes, my mistake
